packages/protocol/contracts/common/MetaTransactionWallet.sol

```diff
@@ -27,11 +27,17 @@ contract MetaTransactionWallet is
   uint256 public nonce;
   address public signer;
 
-  event SignerSet(address signer);
+  event SignerSet(address indexed signer);
   event EIP712DomainSeparatorSet(bytes32 eip712DomainSeparator);
-  event TransactionExecution(address destination, uint256 value, bytes data, bytes returnData);
+  event Deposit(address indexed sender, uint256 value);
+  event TransactionExecution(
+    address indexed destination,
+    uint256 value,
+    bytes data,
+    bytes returnData
+  );
   event MetaTransactionExecution(
-    address destination,
+    address indexed destination,
     uint256 value,
     bytes data,
     uint256 nonce,
```

Comment:
do we think nonce would be useful as an indexed parameter?

Reply:
can you remind me the role of a nonce here?

Reply:
differentiate sequence of transactions signed